### ISSUE TITLE
SC1072,SC1020,SC1073: shellcheck couldn't parse, but Bash is parsing (issue both online & in shellcheck v.0.7.1)

### ISSUE THREAD

**Issue by hippie68** [koalaman/shellcheck#2165]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1072, SC1020, SC1073
- My shellcheck version (`shellcheck --version` or "online"): "online"
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

# $1: message, $2: allowed characters
userinput_char() {
  local char
  while read -e -t 0.01; do :; done
  while [[ $char != ["$2"] ]]; do
    read -p "$1 " -n 1 char
  done
  echo "$char"
}

char=$(userinput_char "Are you sure? [Y]es/[N]o" yn)

```

#### Here's what shellcheck currently says:
 while [[ $char != ["$2"] ]]; do
  ^-- SC1009: The mentioned syntax error was in this while loop.
        ^-- SC1073: Couldn't parse this test expression. Fix to allow more checks.
                          ^-- SC1020: You need a space before the ]].
                          ^-- SC1072: Missing space before ]. Fix any mentioned problems and try again.

#### Here's what I wanted or expected to see:
No errors.

**Comment by celalsahin** [comment]
Any comments on this? I have the same error for a similar check, e.g.

  if [[ $var == \[*] ]]
  ^-- SC1009: The mentioned syntax error was in this if expression.
     ^-- SC1073: Couldn't parse this test expression. Fix to allow more checks.
                    ^-- SC1020: You need a space before the ]].
                    ^-- SC1072: Missing space before ]. Fix any mentioned problems and try again.

**Comment by koalaman** [comment]
This was caused by ShellCheck's overzealous detection of newbie mistakes like `if [[ $var == foo]]`. It now avoids doing this when the word contains a `[`, which will hopefully avoid this egregious error while also remaining helpful. Thanks!

In affected version, one possible (if ugly) workaround is `[[ $char != ["$2"]'' ]]`. The trailing empty quoted string causes the word not to syntactically end with an unquoted `]` and therefore avoids triggering.
